### ISSUE TITLE
Incorrect mime type for JavaScript files

### ISSUE THREAD

**Issue by metrue** [caddyserver/caddy#3405]
## Caddy version
I just cloned the latest master branch and build from source on my Mac.
```
$  ./caddy version
(devel)
```

## OS version
```
$ uname -a
Darwin C02ZL0RJLVDN 18.7.0 Darwin Kernel Version 18.7.0: Mon Feb 10 21:08:45 PST 2020; root:xnu-4903.278.28~1/RELEASE_X86_64 x86_64
```

## Go Version
```
$ go version
go version go1.14 darwin/amd64
```

## Caddyfile
```
:80 {
  encode gzip
  log {
    output file ./access.log
    level DEBUG
  }

  reverse_proxy /api/hello 127.0.0.1:9005
 
  @noneapi {
    not path /api/*
  }
  root * public
  file_server
  rewrite @noneapi /
}
```

## Run command

```
caddy run
```

## Unexpected Response
```
$ curl -s -I  http://localhost/dist/index.js
HTTP/1.1 200 OK
Accept-Ranges: bytes
Content-Length: 2648
Content-Type: text/html; charset=utf-8
Etag: "qa9b6c21k"
Last-Modified: Wed, 13 May 2020 06:25:24 GMT
Server: Caddy
Date: Wed, 13 May 2020 09:29:50 GMT
```

The content type of `index.js` should be `application/javascript` not `text/html`

Does anyone know how to fix it?

**Comment by hairyhenderson** [comment]
This works for me with the released Caddy 2.0.0 binary on macOS:

```console
$ caddy file-server &
[1] 90424
2020/05/13 17:49:06.426 WARN    admin   admin endpoint disabled                                                                                                
2020/05/13 17:49:06.427 INFO    http    server is listening only on the HTTP port, so no automatic HTTPS will be applied to this server {"server_name": "static", "http_port": 80}
2020/05/13 17:49:06.427 INFO    tls     cleaned up storage units
2020/05/13 13:49:06 [INFO][cache:0xc0005a6640] Started certificate maintenance routine
2020/05/13 17:49:06.427 INFO    autosaved config        {"file": "/Users/hairyhenderson/Library/Application Support/Caddy/autosave.json"}
2020/05/13 13:49:06 Caddy 2 serving static files on :80
$ curl -s -I localhost/index.js
HTTP/1.1 200 OK
Accept-Ranges: bytes
Content-Length: 27
Content-Type: application/javascript
Etag: "qaa6o6r"
Last-Modified: Wed, 13 May 2020 17:45:42 GMT
Server: Caddy
Date: Wed, 13 May 2020 17:53:17 GMT
```

Can you try with the released binary, just to make sure it's not an issue with `master`, or your build environment?

**Comment by metrue** [comment]
@hairyhenderson 

Thanks for your confirmation, the root cause is my Caddyfile is not correct, and @francislavoie gave a great [explanation](https://caddy.community/t/could-not-migrate-v1-caddyfile-to-v2/8125/2) in Caddy community, problem solved.
